language/libra-vm/src/libra_transaction_executor.rs

```diff
@@ -687,6 +672,22 @@ impl LibraVM {
     }
 }
 
+fn check_intersection<K: Eq + std::hash::Hash, V: Eq>(
+    lhs: impl Iterator<Item = (K, V)>,
+    rhs: impl Iterator<Item = (K, V)>,
```

Comment:
Would passing a `&HashMap<K, V>` for both of these simplify things (wouldn't need the `.collect`s and the iter/maps above too.

Reply:
Set of <K,V> pairs might also work (+ then you could use the built-in intersection trait).